### ISSUE TITLE
Invalid C code generation due to shadowing

### ISSUE THREAD

**Issue by ashn-dot-dev** [ashn-dot-dev/sunder#122]
Discovered on Sunder commit a48c5b0f8a65e9c8752ae9915037bf45877aa201.

```
struct foo {
    var a: u32;
    var b: s32;
}

func bar(foo: *foo) foo {
    return *foo;
}

func main() void {
    var x = (:foo){.a = 123, .b = 456};
    bar(&x);
}
```

```
$ SUNDER_BACKEND=C sunder-run test.sunder
a.out.tmp.c:61:5: error: must use 'struct' tag to refer to type '__sunder_foo'
    __sunder_foo __sunder_return = {/* uninit */0};
    ^
    struct
a.out.tmp.c:58:38: note: struct '__sunder_foo' is hidden by a non-type declaration of '__sunder_foo' here
__sunder_bar(__sunder_pointer_to_foo __sunder_foo)
                                     ^
1 error generated.
$ SUNDER_BACKEND=nasm sunder-run test.sunder
```

**Comment by ashn-dot-dev** [comment]
Fixed in e317eba.

**Comment by ashn-dot-dev** [comment]
Follup commit in 899fece provides a more comprehensive solution to this issue.
